http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -531,6 +533,15 @@ else if ((cause instanceof NoClassDefFoundError) && cause.getMessage().endsWith(
             value = responseHandler.handle(request, jettyResponse);
         }
         finally {
+            if (jettyResponse != null && jettyResponse.getStatusCode() == 200) {
```

Comment:
This should be done regardless of the status, i.e. for any response that has a body.